### ISSUE TITLE
Create Lambda trigger for RDS EM forwarder

### ISSUE THREAD

**Issue by edjshelton** [terraform-aws-modules/terraform-aws-datadog-forwarders#19]
## Is your request related to a new offering from AWS?

No

## Is this functionality available in the [AWS provider for Terraform](https://registry.terraform.io/providers/hashicorp/aws/latest/docs)? See [CHANGELOG.md](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md), too.
  - Yes ✅: please list the AWS provider version which introduced this functionality

## Is your request related to a problem? Please describe.
The RDS enhanced metrics forwarder module doesn't create the Lambda trigger for sending metrics from the `RDSOSMetrics` CloudWatch log group.

## Describe the solution you'd like.
Add a the trigger as either a subscription filter on the log group or a trigger on the Lambda function.

## Describe alternatives you've considered.
We've had to Terraform the trigger/filter separately, but there's no obvious reason why it can't be included in this module.

## Additional context
Basically step 11 on the installation docs onwards:

https://docs.datadoghq.com/integrations/amazon_rds/?tab=enhanced#installation

**Comment by lukedoesinfra** [comment]
@bryantbiggs - Could you explain the position of not adding the trigger in the above context when you have a second? Thanks
